### DISCUSSION TITLE
Setting the python interpreter in Qodana

### DISCUSSION THREAD

**Discussion by Cazayus** [JetBrains/Qodana#149] [Q&A]
I have a problem similar to #134 and #61.
I want to specify a python version for the analysis.
The problem is that since 2022.3, pyenv is not available in the image anymore.
The solution I almost have is to do something similar to the aforementioned discussions but with conda instead.
It looks like this in the bootstrap section of qodana.yaml :

>  conda init bash
  conda create -y -n mypython python=3.8.5
  conda activate mypython
  python3 -m venv /data/venv
  source /data/venv/bin/activate

It looks good except that "conda activate mypython" fails with "Your shell has not been properly configured to use 'conda activate'."

Is there any way to make this work (or to specify the python version another way, without pyenv) ?

For qodana-jvm, I believe a new field appeared in the yaml and we can set the JDK for the analysis thusly : 
> projectJDK: 17

It is very useful and I guess that all variant of qodana are going to need something similar at some point (setting the PHP version, the C++ compiler, etc)

**Comment by brichbash** [comment]
Hello,

Sorry for the inconvenience,
I understand that configuring some other Python interpreter for Qodana is inconvenient and we strive for making the process easier in the future.
There is an issue [on our tracker](https://youtrack.jetbrains.com/issue/QD-4151/Configure-SDK-for-a-linter-using-qodana.yaml-or-a-flag) that describes exactly your feature request with setting interpreter via `qodana.yaml`. If you have a spare minute, please vote for it to get notified when there are any updates.
As for now, you can try the following workaround:
1. Add `.idea/misc.xml` to your project root with the following content:
```
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="ProjectRootManager" version="2" project-jdk-name="mypython" project-jdk-type="Python SDK" />
</project>
```
2. Download `jdk.table.xml` that is [attached](https://github.com/JetBrains/Qodana/files/10427479/jdk.table.xml.zip) to the message (you might need to edit it, if you want to use other versions of python, now it is configured for Python 3.8.5), you'll need it to mount it to a Docker container with Qodana.
3. Use the following bootstrap in `qodana.yaml`:
```
bootstrap: |
  conda init bash
  conda update -y conda
  conda install -y python=3.8.5
  python --version
  python -m venv venv
  source /data/project/venv/bin/activate
```
4. Run qodana: `qodana scan -v /path/to/jdk.table.xml:/root/.config/idea/options/jdk.table.xml -u root`

**Reply by ReenigneArcher** [reply]
Can we just get `pyenv` back until proper support for specifying the python version is available in the `qodana.yaml`?
